File: Python/BinaryTreePaths/BinaryTreePaths.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution(object):
    # (52ms)
    def binaryTreePaths(self, root):
        """
        :type root: TreeNode
        :rtype: List[str]
        """
        def helper(root, ans, now):
            if not (root.left or root.right):
                ans.append(now + str(root.val))
                return
            now = now + str(root.val) + "->"
            if root.left:
                helper(root.left, ans, now)
            if root.right:
                helper(root.right, ans, now)

        ans = []
        if root:
            helper(root, ans, '')
        return ans

File: JavaScript/BinaryTreePaths/BinaryTreePaths.js
/**
 * Created by Phoenix-Z on 2017/4/8.
 */
// Definition for a binary tree node.
function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
// (186ms)
var binaryTreePaths = function(root) {
    var helper = function (root, ans, now) {
        if(root.left === null && root.right === null) {
            ans.push(now + root.val);
            return;
        }
        now += root.val + "->";
        if(root.left !== null)
            helper(root.left, ans, now);
        if(root.right !== null)
            helper(root.right, ans, now);
    };
    var ans = [];
    if(root !== null)
        helper(root, ans, "");
    return ans;
};
